### PR TITLE
fix: objectId casting now working

### PR DESCRIPTION
**Pull request by giuliowaitforitdavide**
## Pull Request Type

- [X] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [X] Documentation content changes
- [ ] Other... Please describe:

## Description

The getList was returing an error `Cannot read properties of undefined, reading .toString()` when try to cast ObjectIds

## PR Checklist

- [X] The commit message follows our guidelines included in the [CONTRIBUTING.md](../CONTRIBUTING.md#how-to-submit-a-pr)
- [X] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

**Review by ThisIsDemetrio — APPROVED** [review]
lgtm

**Review by hiimjako — CHANGES_REQUESTED** [review]
It seems fine, but if you try to look at the generated swagger you will see that it some fields it shows: `Unknown Type: string,object` or `Unknown Type: array,string`.
Can you try to give it a look? It seems an error related to JSONschemaGenerator and/or ajv configuration.

**Comment by danibix95** [comment]
> It seems fine, but if you try to look at the generated swagger you will see that it some fields it shows: `Unknown Type: string,object` or `Unknown Type: array,string`. Can you try to give it a look? It seems an error related to JSONschemaGenerator and/or ajv configuration.

@giuliowaitforitdavide

**Comment by giuliowaitforitdavide** [comment]
> It seems fine, but if you try to look at the generated swagger you will see that it some fields it shows: `Unknown Type: string,object` or `Unknown Type: array,string`. Can you try to give it a look? It seems an error related to JSONschemaGenerator and/or ajv configuration.

Fixed

File: lib/schemaGetters.js [hunk 1]
```diff
@@ -171,16 +171,16 @@ function validateDateSchemaCompatibility(jsonSchema) {
...
         {
           type: 'string',
           pattern: '^\\d{4}-\\d{2}-\\d{2}(T\\d{2}:\\d{2}:\\d{2}(\\.\\d{1,3})?(Z|[+-]\\d{2}:\\d{2}))?$',
           description: '"date-time" according with https://tools.ietf.org/html/rfc3339#section-5.6',
           examples: [EXAMPLE_DATE],
         },
+        {
+          type: 'null',
+          nullable: jsonSchema.schema.nullable,
+        },
```
[the remark this answers is not in the document]
Comment (giuliowaitforitdavide) [reply lib/schemaGetters.js]: Fixed

**Comment by hiimjako** [comment]
Still have the same problem on `ObjectId`, `Date` and `Array of object` field. The fix should be something similar to the previous ones.

**Review by hiimjako — APPROVED** [review]
LGTM, good job!

**Pull request merged**